must-gather/gather

```diff
@@ -0,0 +1,114 @@
+# distributed under the license is distributed on an "as is" basis,
+# without warranties or conditions of any kind, either express or implied.
+# see the license for the specific language governing permissions and
+# limitations under the license.
+#
+
+set -eu -o pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+source "${SCRIPT_DIR}"/common
+
+print_usage() {
+    echo -e "Usage: /usr/bin/gather <gather-collection-path>"
+}
+
+get_kepler_instance() {
+  log "getting kepler instance"
+  run oc get keplers.kepler.system.sustainable.computing.io kepler -oyaml "$BASE_COLLECTION_PATH/kepler.yaml"
+}
+
+get_kepler_events() {
+  log "getting $KEPLER_NS events"
+  run oc -n "$KEPLER_NS" get events "$BASE_COLLECTION_PATH/$KEPLER_NS""_events"
+}
+
+get_kepler_daemon_set() {
+  KEPLER_EXPORTER_DS="kepler-exporter-ds"
+  log "getting kepler exporter daemonset"
+  run oc -n "$KEPLER_NS" get ds $KEPLER_EXPORTER_DS -oyaml "$BASE_COLLECTION_PATH/$KEPLER_EXPORTER_DS"".yaml"
+}
+
+get_kepler_config_map() {
+  KEPLER_EXPORTER_CM="kepler-exporter-cm"
+  log "getting kepler exporter config map"
+  run oc -n "$KEPLER_NS" get cm $KEPLER_EXPORTER_CM -oyaml "$BASE_COLLECTION_PATH/$KEPLER_EXPORTER_CM"".yaml"
+}
+
+get_kepler_sa() {
+  KEPLER_EXPORTER_SA="kepler-exporter-sa"
+  log "getting kepler exporter service account"
+  run oc -n "$KEPLER_NS" get serviceaccount $KEPLER_EXPORTER_SA -oyaml "$BASE_COLLECTION_PATH/$KEPLER_EXPORTER_SA"".yaml"
+}
+
+get_kepler_scc() {
+  KEPLER_EXPORTER_SCC="kepler-exporter-scc"
+  log "getting kepler exporter service account"
+  run oc get scc "$KEPLER_EXPORTER_SCC" -oyaml "$BASE_COLLECTION_PATH/$KEPLER_EXPORTER_SCC"".yaml"
+}
+
+gather_kepler_exporter_info() {
+  KEPLER_PODS=$(oc -n "$KEPLER_NS" get pods -oname)
+  for pod in $KEPLER_PODS
+  do
+	pod=$(echo "$pod" | awk -F '/' '{print $2}')
+	log "running gather script for kepler pod: $pod"
+	"$SCRIPT_DIR"/gather-kepler-exporter-info "$BASE_COLLECTION_PATH" "$pod"
+  done
+}
+
+gather_olm_info(){
+  log "running gather script for olm"
+  "$SCRIPT_DIR"/gather-olm-info "$BASE_COLLECTION_PATH"
+}
+
+gather_kepler_operator_info() {
+  log "getting kepler-operator info"
+  "$SCRIPT_DIR"/gather-kepler-operator-info "$BASE_COLLECTION_PATH"
+}
+
+main(){
+  case ${1-} in
+	--help | -h )
+	  print_usage
+	  exit 1
+  esac
+
+  BASE_COLLECTION_PATH="${1:-/must-gather}"
+  export LOGFILE_PATH="${BASE_COLLECTION_PATH}/${LOGFILE_NAME}"
+
+  mkdir -p "${BASE_COLLECTION_PATH}"
+  cd "${BASE_COLLECTION_PATH}"
+  echo -e "must-gather logs are located at: '${LOGFILE_PATH}'"
+
+  mkdir -p "${BASE_COLLECTION_PATH}/cache-dir"
+  export KUBECACHEDIR=${BASE_COLLECTION_PATH}/cache-dir
+
+  echo "powermon must-gather started..."
+  get_kepler_instance
+  get_kepler_events
+  get_kepler_daemon_set
+  get_kepler_config_map
+  get_kepler_sa
+  get_kepler_scc
+  gather_kepler_exporter_info
+  gather_olm_info
```

Comment:
shouldn't we begin with capturing olm and operator info?

Reply:
In the test scenario, I installed only the operator (imagine the operator installation failed) and ran the must gather. 
```
❯ less after-install/quay-io-sthaha-kepler-operator-sha256-df656da3b9f7a0c4310f8a44cb7b11eb790a7be43794e67c1343e934953f6478/gather-debug.log
2023-10-05 01:23:52 getting kepler instance
2023-10-05 01:23:52 oc get keplers.kepler.system.sustainable.computing.io kepler -oyaml
Error from server (NotFound): keplers.kepler.system.sustainable.computing.io "kepler" not found
```

And thats all it captures

Reply:
the script is not stateful.

Reply:
let me rephrase, in the scenario where the operator fails to install, i.e. no `kepler` CRD in cluster, what information should we capture?

Reply:
if script executes all its steps, we will capture olm info.